src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,216 @@
+        final DetailAST outerNotEqualNode = getExprChildNode(ast, TokenTypes.NOT_EQUAL);
+        boolean error = false;
+        if (isNullTest(outerNotEqualNode)) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST instanceOfNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.LITERAL_INSTANCEOF);
+                if (instanceOfNode != null) {
+                    error = isTextEqual(instanceOfNode, outerNotEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    private static boolean checkOuterInstanceOf(DetailAST ast) {
+        final DetailAST outerInstanceOfNode =
+                getExprChildNode(ast, TokenTypes.LITERAL_INSTANCEOF);
+        boolean error = false;
+        if (outerInstanceOfNode != null) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST notEqualNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.NOT_EQUAL);
+                if (isNullTest(notEqualNode)) {
+                    error = isTextEqual(outerInstanceOfNode, notEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    /**
+     * To get specific Node that is child of {@link TokenTypes#EXPR} Node.
+     *
+     * @param ast DetailAST
+     * @param token int
+     * @return DetailAST
+     */
+    private static DetailAST getExprChildNode(DetailAST ast, int token) {
+        return ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(token);
+    }
+
+    /**
+     * True if nodes have a {@link TokenTypes#LITERAL_NULL}, false otherwise.
+     *
+     * @param ast DetailAST
+     * @return boolean
+     */
+    private static boolean isNullTest(DetailAST ast) {
+        return ast != null
+                && ast.findFirstToken(TokenTypes.LITERAL_NULL) != null;
+    }
+
+    /**
+     * Gets nested {@link TokenTypes#LITERAL_IF} node.
+     *
+     * @param ast DetailAST
+     * @return nestedIfNode DetailAST
+     */
+    private static DetailAST getNestedIfNode(DetailAST ast) {
+        final DetailAST slistNode = ast.findFirstToken(TokenTypes.SLIST);
+        final DetailAST nestedIfNode;
+        if (slistNode == null) {
+            nestedIfNode = ast.findFirstToken(TokenTypes.LITERAL_IF);
+        }
+        else {
+            nestedIfNode = slistNode.findFirstToken(TokenTypes.LITERAL_IF);
+        }
+        return nestedIfNode;
+    }
+
+    /**
+     * True if nodes have same texts, false otherwise.
+     *
+     * @param instanceOfNode DetailAST
+     * @param notEqualNode DetailAST
+     * @return error boolean
+     */
+    private static boolean isTextEqual(DetailAST instanceOfNode, DetailAST notEqualNode) {
+        return getFullText(instanceOfNode).equals(getFullText(notEqualNode));
```

Comment:
you can use `instanceOfNode != null && isNullTest(notEqualNode)` here and remove these conditions from other methods

Reply:
done